src/modelfree/shaping_wrappers.py

```diff
@@ -0,0 +1,215 @@
+from collections import defaultdict
+import json
+import os
+
+import numpy as np
+from stable_baselines import logger
+from stable_baselines.common.base_class import BaseRLModel
+from stable_baselines.common.vec_env import VecEnvWrapper
+
+from modelfree import envs  # noqa - needed to register sumo_auto_contact
+from modelfree.scheduling import LinearAnnealer
+
+
+class RewardShapingVecWrapper(VecEnvWrapper):
+    """A more direct interface for shaping the reward of the attacking agent."""
+    def __init__(self, venv, agent_idx, shaping_params, batch_size, reward_annealer=None):
+        super().__init__(venv)
+        # structure: {'sparse': {k: v}, 'dense': {k: v}, **kwargs}
+        self.shaping_params = shaping_params
+        self.reward_annealer = reward_annealer
+        self.counter = 0
+        self.batch_size = batch_size
+        self.agent_idx = agent_idx
+
+        self.ep_rew_dict = defaultdict(list)
+        self.step_rew_dict = defaultdict(lambda: [[] for _ in range(self.num_envs)])
+
+    @classmethod
+    def get_shaping_params(cls, shaping_params, env_name):
+        path_stem = 'experiments/rew_configs/'
+        default_configs = {
+            'multicomp/SumoHumans-v0': 'default_hsumo.json',
+            'multicomp/SumoHumansAutoContact-v0': 'default_hsumo.json'
+        }
+        assert env_name in default_configs, \
+            "No default reward shaping config found for env_name:{}".format(env_name)
+        fname = os.path.join(path_stem, default_configs[env_name])
+        final_shaping_params = json.load(open(fname))
```

Comment:
This will leak file descriptors, it should be:
```
with open(fname) as f:
    final_shaping_params = json.load(f)
```

This is also fragile since `fname` depends on your current directory. I suggest instead adding the configs to `setup.py` using the `data_files` keyword, see https://docs.python.org/3/distutils/setupscript.html#installing-additional-files, and then using `pkgresources` to load it.